dialogue-clients/src/main/java/com/palantir/dialogue/clients/ChannelCache.java

```diff
@@ -189,9 +193,11 @@ public String toString() {
     interface ChannelCacheKey extends AugmentClientConfig {
         ServiceConfiguration serviceConf();
 
+        Optional<ExecutorService> blockingExecutor();
+
         String channelName();
 
-        Optional<ExecutorService> blockingExecutor();
+        OptionalInt overrideHostIndex();
```

Comment:
I'm a little worried about this fields (and the blockingExecutor for that matter) resulting in cache misses. Would bad things happen if we annotated `@Auxiliary`?

Reply:
Don't think we want to do that, because if we ask for something with the hostIndex set to 6 and get a cache hit with the hostIndex set to 0 then we'll get confusing metrics.

However I think the cache misses are not very serious here because this class has both the `channelCache` (full of DialogueChannels) and the `apacheCache` (full of `ApacheHttpClientChannels.CloseableClient`s). We'll still get cache hits on the apache clients, so I don't think this PR will result in problems with connection pools.